### PR TITLE
FIX: `Topic.qa_enabled` raises an error for PM topics.

### PR DESCRIPTION
**Pull request by tgxworld**

### CODE REVIEW

File: spec/models/topic_spec.rb [hunk 1]
```diff
@@ -77,126 +77,126 @@
...
-      end
+  describe '.qa_enabled' do
+    it 'should return false if topic is blank' do
+      expect(Topic.qa_enabled(nil)).to eq(false)
+    end
 
-      it 'should return false if disabled' do
-        SiteSetting.qa_enabled = false
+    it 'should return false for a PM' do
+      expect(Topic.qa_enabled(Fabricate(:private_message_topic))).to eq(false)
```
Comment (tgxworld) [inline spec/models/topic_spec.rb:160]: This is the test I added among all the indentation changes.

Comment (martin-brennan) [reply spec/models/topic_spec.rb:160]: Really don't understand why GitHub doesn't just default to not showing whitespace changes. Totally irrelevant in most cases.

File: extensions/topic_extension.rb [hunk 2]
```diff
@@ -95,7 +95,8 @@ def qa_votes(topic, user)
 
       def qa_enabled(topic)
         return false unless SiteSetting.qa_enabled
-        return false if !topic || topic.id == topic.category.topic_id
+        return false if !topic
+        return false if topic.category && topic.category.topic_id == topic.id
```
Comment (martin-brennan) [inline extensions/topic_extension.rb:99]:
```suggestion
        return false if topic.category&.topic_id == topic.id
```

**Review by martin-brennan — APPROVED** [review]

File: extensions/topic_extension.rb (hunk 2 above)
Comment (tgxworld) [reply extensions/topic_extension.rb:99]: I intentionally chose not to do this so that we don't execute the comparison.

**Pull request merged**